src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs

```diff
@@ -1355,5 +1374,120 @@ private void VerifyNotepadMainWindowTitle(Process process, string filename)
         }
+
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))] // No Notepad on Nano
+        [OuterLoop("Launches notepad")]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public void StartInfo_LoadUserProfile_And_UseCredentialsForNetworkingOnly_AreIncompatible()
+        {
+            ProcessStartInfo info = new ProcessStartInfo
+            {
+                UseShellExecute = true,
+                LoadUserProfile = true,
+                UseCredentialsForNetworkingOnly = true,
+                FileName = "notepad.exe",
+                Arguments = null,
+                WindowStyle = ProcessWindowStyle.Minimized
+            };
+
+            Assert.Throws<ArgumentException>("startInfo", () =>
+            {
+                using (var process = Process.Start(info))
+                {
+                    Assert.False(process != null, $"Process started despite incompatible options {nameof(info.LoadUserProfile)} and {nameof(info.UseCredentialsForNetworkingOnly)} were enabled");
+                }
+            });
+        }
+
+        [ConditionalFact(nameof(IsAdmin_IsNotNano_RemoteExecutorIsSupported))] // Nano has no "netapi32.dll", Admin rights are required
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [OuterLoop("Requires admin privileges")]
+        public void TestUserNetworkCredentialsPropertiesOnWindows()
+        {
+            string testFilePath = GetTestFilePath();
+            Assert.False(string.IsNullOrWhiteSpace(testFilePath), $"Path to test file should not be empty: {testFilePath}");
+
+            string testFilePathRoot = Path.GetPathRoot(testFilePath);
+            string testFileUncPath = $"\\\\{Environment.MachineName}\\{testFilePath.Substring(0, testFilePathRoot.IndexOf(":"))}$\\{testFilePath.Replace(testFilePathRoot, "")}";
+            string testFileContent = Guid.NewGuid().ToString();
+            File.WriteAllText(testFilePath, testFileContent);
+
+            using WindowsTestAccount accountWithoutFilePermission = new WindowsTestAccount("testForDotnetRuntime");
+
+            const string username = "testForDotnetNetwork";
+            string password = Convert.ToBase64String(RandomNumberGenerator.GetBytes(33)) + "_-As@!%*(1)4#2";
+
+            uint removalResult = Interop.NetUserDel(null, username);
+            Assert.True(removalResult == Interop.ExitCodes.NERR_Success || removalResult == Interop.ExitCodes.NERR_UserNotFound);
+
+            Interop.NetUserAdd(username, password);
+
+            bool hasStarted = false;
+            Process p = null;
+            string workingDirectory = null;
+
+            try
+            {
+                // Schedule a process to see what env vars it gets.  Have it write out those variables
+                // to its output stream so we can read them.
+                p = CreateProcess(arg =>
+                {
+                    Console.Write(File.ReadAllText(arg));
```

Comment:
would it be possible to extend this test with checking that using `testFilePath` to read the file throws?

Reply:
Oh, interesting. I actually hadn't thought about that case, but that's a valid point. Will have a look!